api/internal/localizer/localizer.go

```diff
@@ -361,9 +357,26 @@ func (lc *localizer) localizeBuiltinPlugins(kust *types.Kustomization) error {
 	return nil
 }
 
-// loadResource tries to load resourceEntry as a file path or inline.
-// On success, loadResource returns the loaded resource map and whether resourceEntry is a file path.
-func (lc *localizer) loadResource(resourceEntry string) (resmap.ResMap, bool, error) {
+// localizeK8sResource returns the localized file path if resourceEntry is a
+// file containing a kubernetes resource.
+// localizeK8sResource returns the empty string if resourceEntry is an inline
+// resource.
+func (lc *localizer) localizeK8sResource(resourceEntry string) (string, error) {
```

Comment:
I renamed `localizeResource` and `loadResource` to distinguish the use of "resource" to refer to a kubernetes resource here from its referral to the kustomize `resources` field in #4912: https://github.com/kubernetes-sigs/kustomize/pull/4912/files#diff-25faa786bf35469d391d5eef99ed1323623023442b177c1ed3e07f3e6709dac6R261